### ISSUE TITLE
Provider not supporting M1 Mac (darwin_arm64)

### ISSUE THREAD

**Issue by roshan8** [elastic/terraform-provider-ec#414]
- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed
- [x] I am reporting the issue to the correct repository (for multi-repository projects)

## Expected Behavior
I should be able to perform terraform apply command on M1 Mac(darwin_arm64)

## Current Behavior
I get the following error when I try `terraform init`
```
│ Error: Incompatible provider version
│ Provider registry.terraform.io/elastic/ec v0.3.0 does not have a package available for your current platform, darwin_arm64.
│ Provider releases are separate from Terraform CLI releases, so not all providers are available for all platforms. Other versions of this provider may have different platforms supported.
```

My terraform version:
```
> terraform version
Terraform v1.0.6
on darwin_arm64
```

Provider version:
```
  required_providers {
    ec = {
      source  = "elastic/ec"
      version = "0.3.0"
    }
  }
```

## Additional info:
I do see a [PR](https://github.com/elastic/terraform-provider-ec/pull/315) that got merged in May. However, I don't see the `darwin_arm64` package in the recent [release](https://github.com/elastic/terraform-provider-ec/releases)

**Comment by PatrickHeneise** [comment]
Maybe the CI can't build darwin/arm64?

`go tool dist list`

Don't see why it shouldn't build/release otherwise. Have the same issue now.

**Comment by cgroschupp** [comment]
<img width="818" alt="grafik" src="https://user-images.githubusercontent.com/6190136/147939544-f451e2d2-f928-4007-9135-4b341fc8f56c.png">
The release v0.2.1 already support arm64. can anyone look inside the jenkins release pipeline?

**Comment by cgroschupp** [comment]
<img width="818" alt="grafik" src="https://user-images.githubusercontent.com/6190136/147939544-f451e2d2-f928-4007-9135-4b341fc8f56c.png">
The release v0.2.1 already support arm64. can anyone look inside the jenkins release pipeline?

**Comment by artemnikitin** [comment]
> I do see a PR that got merged in May. However, I don't see the darwin_arm64 package in the recent release

Yes, the fix was merged. The next release should have Mac M1 packages 🤞

**Comment by vhatsura** [comment]
Hey, @artemnikitin

Thanks for the update. Do you know any timeline for the release of a new version?

**Comment by jbielick** [comment]
Is anyone available to do a release? Sounds like that will fix issues for folks on 0.3.0 and darwin/arm64.

As a workaround, one might be able to downgrade to 0.2.1 as there is an arm64 binary for that release.

**Comment by dimuon** [comment]
The releases starting from 0.4.0 contain builds for `darwin_arm64` - https://github.com/elastic/terraform-provider-ec/releases/download/v0.4.0/terraform-provider-ec_0.4.0_darwin_arm64.zip and https://github.com/elastic/terraform-provider-ec/releases/download/v0.4.1/terraform-provider-ec_0.4.1_darwin_arm64.zip
